src/compiler/checker.ts

```diff
@@ -4306,9 +4306,10 @@ namespace ts {
             if (strictNullChecks && declaration.initializer && !(getFalsyFlags(checkExpressionCached(declaration.initializer)) & TypeFlags.Undefined)) {
                 type = getTypeWithFacts(type, TypeFacts.NEUndefined);
             }
-            return declaration.initializer ?
+            type = declaration.initializer ?
                 getUnionType([type, checkExpressionCached(declaration.initializer)], /*subtypeReduction*/ true) :
                 type;
+            return shouldUseLiteralType(declaration) ? type : getWidenedLiteralType(type);

```

Comment:
What are the cases that are affected by this change?

Reply:
You can see the changes at https://github.com/weswigham/TypeScript/pull/1/files